devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -489,9 +493,13 @@ private void restoreTerminalState() {
     private void handleAutoCompile() throws MojoExecutionException {
         //we check to see if there was a compile (or later) goal before this plugin
         boolean compileNeeded = true;
+        boolean indexClassNeeded = true;
         boolean testCompileNeeded = true;
         boolean prepareNeeded = true;
         boolean prepareTestsNeeded = true;
+
+        String jandexGoalPhase = getGoalPhaseOrNull(ORG_JBOSS_JANDEX, JANDEX_MAVEN_PLUGIN, "jandex");
```

Comment:
If `jandexGoalPhase` is null, `indexClassNeeded` could be initialized to `false` right away, right?
e.g.
````
boolean indexClassNeeded = jandexGoalPhase != null;
````

Reply:
right, I pushed the fix.